nsxt/data_source_nsxt_transport_node.go

```diff
@@ -0,0 +1,100 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+func dataSourceNsxtTransportNode() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtTransportNodeRead,
+
+		Schema: map[string]*schema.Schema{
+			"id":           getDataSourceIDSchema(),
+			"display_name": getDataSourceExtendedDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"node_type": {
```

Comment:
Would it make sense to filter by node type as well?

Reply:
Would that produce a unique result, other than in testing environments?

Reply:
Seems to me like such filtering could be helpful, but we can wait for the ask

Reply:
Maybe it's better to wait and see if we want to convert this into nsxt_edge_transport_node as discussed? Or maybe we change only the resource?

Reply:
I don't really have a preference, either is fine I think